WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java

```diff
@@ -476,7 +476,12 @@ public void onClick(View view) {
                         break;
                     default:
                         if (mOnPostButtonClickListener != null) {
-                            mOnPostButtonClickListener.onPostButtonClicked(buttonType, post);
+                            PostModel latestPost = mPostStore.getPostByLocalPostId(post.getId());
+                            if (latestPost == null) {
+                                loadPosts(LoadMode.FORCED);
+                                return;
+                            }
+                            mOnPostButtonClickListener.onPostButtonClicked(buttonType, latestPost);
```

Comment:
Good point - there was another case or two I missed too. I moved the post reload code to `onPostButtonClicked`, where it should catch all cases, in 6b0ce7e.